### PR TITLE
not traverse, but sequence silly

### PR DESCRIPTION
**Pull request by DrBoolean**
#113 pointed out that I'd left this as the previous traverse implementation. This updates the example

### CODE REVIEW

File: id.js [hunk 1]
```diff
@@ -33,8 +33,8 @@ Id.prototype.ap = function(b) {
 };
 
 // Traversable
-Id.prototype.sequence = function(f, of) {
-    return f(this.value).map(function(y){ return new Id(y); });
+Id.prototype.sequence = function(of) {
+    return this.value.map(function(y){ return new Id(y); });
```
Comment (davidchambers) [inline id.js]: Let's use `Id.of` in place of `function(y){ return new Id(y); }`. :)

**Comment by davidchambers** [comment]
I was about to open a pull request for this myself, but paused when I realized the `of` parameter is ignored. Can you explain why this is so? Is it necessary for fold-based implementations, but not for single-value containers such as Id?

**Comment by DrBoolean** [comment]
Hi @davidchambers 

I put a quick comment in there for others as well. The `of` must be provided for types which might fail. A good example would be Maybe:

```js
var j = Just([2])
j.sequence(Array.of)
// [Just(2)]

var n = Nothing
n.sequence(Array.of)
// [Nothing]
```

In the second case, if we didn't provide `Array.of`, we wouldn't know what type to return.

Most instances do not need to take advantage of the `of` argument, but it must be provided to keep a standard api.

**Comment by davidchambers** [comment]
Thanks for the explanation. :)

:deciduous_tree:

**Pull request merged**